Linkta-org/core
# Logging imrovements; scheduled backups

This PR makes various improvements to server-side logging.

- Logger configuration is moved out of server.ts and into its own file.
- In production mode, logs are output to both stdout and to a local file.
- Log files are rolled daily, the date is included in the filenames, and the files are compressed via Gzip.
- In production mode, a cron job periodically stores the compressed log files to the database.
- Prevent file duplication on the database
- Implement a log retention policy, e.g., "Retain previous thirty days"
- Create an endpoint to download log files
- Create a "back office" UI for downloading logs



Fixes # N/A

## Type of change





## How Can this be tested? Testing Plan to review this PR



server/utils/loggerConfig.ts and loggerScheduledTasks.ts

tl;dr - Just request a demo.

- Manipulate the date code in loggerConfig / loggerProd.appenders.file.pattern to roll the logs every second or every minute.
- Manipulate the cron expressions loggerScheduledTasks / uploadLogs() and evictLogs() to run the tasks frequently and evict files created recently
- Set server/.env/NODE_ENV variable to 'production'
- Create LOGGER_DB_URI variable with 'log_backup' as the db reference

![image](https://github.com/user-attachments/assets/0827496b-c10d-4ff3-a2b0-85d6e5410822)


What is the expected output?

- In production mode, the server should store INFO level logs in a local file
- The local log file should be rolled daily 
- The current "hot" log file and any other local log files should be persisted to the database daily
- Log files aged more than 30 days should be deleted from the database daily
- Log files from a specified date range can be downloaded via the admin UI

## Checklist:




## Additional Information:
